www/js/app.js

```diff
@@ -1255,7 +1254,12 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
 
         // Replaces ZIM-style URLs of img, script, link and media tags with a data-kiwixurl to prevent 404 errors [kiwix-js #272 #376]
         // This replacement also processes the URL to remove the path so that the URL is ready for subsequent jQuery functions
-        htmlArticle = htmlArticle.replace(regexpTagsWithZimUrl, '$1data-kiwixurl$2');
+        // If we are in a new-style ZIM archive with a single namespace for user content, then that namespace is added to the URL
+        htmlArticle = htmlArticle.replace(regexpTagsWithZimUrl, function(m0, m1, m2, url) {
+            // If it's a new ZIM URL without namespace, add the C namespace (=Content)
+            url = /^[-IJ]/.test(url) ? url : 'C/' + url;
```

Comment:
Do we need to calculate the relative path to the asset before adding the `C` namespace? For hyperlinks, we do ned to calculate relative paths, but for assets, it depends whether they can now exisst in subdirectories of an article. Until recently, all assets would be in a separate namespace `-`, `I` or `J`, therefore the relative path to that namespace was irrelevant and could be discarded. Without knowing the structure of the ZIM, i.e. whether it can place assets such as stylesheets as subdirectories of articles, we don't know whether we need to do a relative path calculation here. @kelson42 do you have any idea about this? (NB in Service Worker mode this will "just work", but not in jQuery mode.)

Reply:
For example, if we have an article on China, let's say `../China`, could we potentially have assets appearing like `../China/the_great_wall.webp`? Or with the Wikivoyage `../Singapore/Chinatown`, could an asset appear as `Tooth_Relic_Temple.png.webp`, so we have to calculate that the URL to the asset is `C/Singapore/Chinatown/Tooth_Relic_Temple.png.webp`?

Reply:
I've now written some "universal" code that calculates ZIM asset URLs relative to the page's ZIM URL. This appears to work indifferently on old-style URLs with namespace and on new-style URLs without namespace. (So long as the relative URLs are written correctly - see  https://github.com/kiwix/kiwix-js/issues/597#issuecomment-767828723 for a (no-namespace) case where the URLs appear to be have written incorrectly.